plots/ab-screenshot/analyze.js

```diff
@@ -0,0 +1,211 @@
+/**
+ * @license
+ * Copyright 2017 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+'use strict';
+
+const fs = require('fs');
+const path = require('path');
+
+const args = require('yargs').argv;
+
+const Metrics = require('../../lighthouse-core/lib/traces/pwmetrics-events');
+
+const constants = require('../constants');
+const utils = require('../utils');
+
+/**
+ * Do a comparison analysis of two batches of runs. This helps you compare how a change in
+ * lighthouse can affect perf results in real-world sites.
+ *
+ * Example usage:
+ * node analyze.js -a ./out-first -b ./out-second
+ */
+function main() {
+  const outPathA = args.a;
+  const outPathB = args.b;
+
+  if (!utils.isDir(outPathA) || !utils.isDir(outPathB)) {
+    console.log('ERROR: Make sure both -a and -b point to valid paths'); // eslint-disable-line no-console
+    console.log('a: ', outPathA); // eslint-disable-line no-console
+    console.log('b: ', outPathB); // eslint-disable-line no-console
+    return;
+  }
+
+  const aggregatedScreenshots = {
+    data: aggregate(outPathA, outPathB),
+    a: outPathA,
+    b: outPathB
+  };
+
+  if (!utils.isDir(constants.OUT_PATH)) {
+    fs.mkdirSync(constants.OUT_PATH);
+  }
+  const outFilePath = path.resolve(constants.OUT_PATH, 'screenshotsComparison.js');
+  fs.writeFileSync(
+    outFilePath,
+    `var aggregatedScreenshots = ${JSON.stringify(aggregatedScreenshots, undefined, 2)}`
+  );
+  console.log('Wrote output to:', outFilePath); // eslint-disable-line no-console
+}
+
+main();
+
+/**
+ * Aggregates the results from two out paths.
+ * Note: only the first run for each site of each batch is used.
+ * @param {string} outPathA
+ * @param {string} outPathB
+ * @return {!AggregatedScreenshots}
+ */
+function aggregate(outPathA, outPathB) {
+  const results = [];
+
+  fs.readdirSync(outPathA).forEach(siteDir => {
+    const sitePathA = path.resolve(outPathA, siteDir);
+    const sitePathB = path.resolve(outPathB, siteDir);
+
+    // Skip a site if it's not in both batches.
+    if (!utils.isDir(sitePathB)) {
+      return;
+    }
+    const siteScreenshotsComparison = {
+      siteName: siteDir,
+      runA: analyzeSingleRunScreenshots(sitePathA),
+      runB: analyzeSingleRunScreenshots(sitePathB)
+    };
+    results.push(siteScreenshotsComparison);
+  });
+
+  return results;
+}
+
+/**
+ * Analyzes the screenshots for the first run of a particular site.
```

Comment:
hmmm I guess the order doesn't really matter then if it's not an aggregation of any sort? I guess we could just pull any random folder huh...sorting just for consistency?